projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -22,87 +37,149 @@ export class PendingRequestComponent implements OnInit, OnChanges {
     { field: "at", order: -1 },
     { field: "member_id", order: 1 },
   ];
-  prevSortMeta = "-at member_id";
-
-  @Output() onAccept = new EventEmitter<any>();
-  @Output() onReject = new EventEmitter<any>();
-
-  groupSwitch = [
-    {
-      label: "This group only",
-    },
-    {
-      label: "All subgroups",
-    },
-  ];
+  prevSortMeta: string = "-at member_id";
+
+  requestLoading: boolean = false;
+  requestAction: string = "";
+  selection = [];
+
+  reloadData(sortBy = []) {
+    this.selection = [];
+    if (sortBy.length === 0) {
+      sortBy = this.prevSortMeta.split(" ");
+    }
+    this.groupService
+      .getManagedRequests(this.id, sortBy)
+      .subscribe((reqs: PendingRequest[]) => {
+        this.requests = [];
+
+        this.requests = reqs.map((req) => {
```

Comment:
As sort and updates retrigger a call on the backend, I am not sure there is a benefit to store these requests. Is there?

Reply:
Yes, to pass the data to the grid, I need to store it to a variable.
And select all functionality needs this

Reply:
You can pass data to the grid via an observable, it makes much more sense with RxJS (you should have a smaller state as possible).
For the select all, it seems you do not have to implement anything and the default behavior work.